spec/src/main/asciidoc/chapters/language/expressions.adoc

```diff
@@ -423,11 +423,29 @@ Within the pattern, `_` matches any single character, and `%` matches any sequen
 
 The equality and inequality operators are `=`, `&lt;&gt;`, `&lt;`, `&gt;`, `&lt;=`, `&gt;=`.
 
+The operands of an equality or inequality operator must have the same type,
+ and it must be an atomic type considered _comparable_ by the implementation of Jakarta Query.
+Any type with a <<natural-order,natural order>> must be treated as a comparable type, and the equality and inequality operators must be interpreted in a way which is consistent with the natural order.
+
+[cols="15,~"]
+|===
+| Operator | Interpretation in terms of natural order
+
+| `x < y` | Satisfied if `x` occurs before `y` in the natural order
+| `x > y` | Satisfied if `x` occurs after `y` in the natural order
+| `x <= y` | Satisfied if `x>y` is not satisfied
+| `x >= y` | Satisfied if `x<y` is not satisfied
+| `x = y` | Satisfied if neither `x>y` nor `x<y` is satisfied
```

Comment:
@otaviojava This is the closest we come to a definition of equality in Jakarta Query. Note that it's by definition consistent with the natural order.

I will add an admonition pointing out that `=` is different to `BigDecimal.equals()`.